packages/ui-components/src/components/UIToggle/UIToggle.tsx

```diff
@@ -92,13 +166,60 @@ const COLORS = {
  * @extends {React.Component<IToggleProps, {}>}
  */
 export class UIToggle extends React.Component<UIToggleProps, {}> {
+    private toggleRootRef: React.RefObject<HTMLDivElement>;
     /**
      * Initializes component properties.
      *
      * @param {UIToggleProps} props
      */
     public constructor(props: UIToggleProps) {
         super(props);
+        this.toggleRootRef = React.createRef<HTMLDivElement>();
+        this.handleChange = this.handleChange.bind(this);
+        this.replaceThumbWithIcon = this.replaceThumbWithIcon.bind(this);
+    }
+
+    /**
+     * Lifecycle method called immediately after a component is mounted.
+     * Executes initialization logic such as DOM manipulations or fetching data.
+     *
+     * @returns {void} This method does not return a value.
+     */
+    componentDidMount() {
+        this.replaceThumbWithIcon();
+    }
+
+    /**
+     * Handles the change event triggered by the user interaction.
+     *
+     * @param {React.MouseEvent<HTMLElement>} event - The mouse event object associated with the interaction.
+     * @param {boolean} [checked] - An optional parameter indicating the current state of the interaction.
+     * @returns {void} This method does not return a value.
+     */
+    handleChange(event: React.MouseEvent<HTMLElement>, checked?: boolean) {
+        this.replaceThumbWithIcon(checked);
+        this.props.onChange?.(event, checked);
+    }
+
+    /**
+     * Replaces the thumb element of a toggle switch with an icon based on the toggle's state.
+     *
+     * @param {boolean} [checked] Optional. Represents the state of the toggle switch. If not provided, it checks `defaultChecked` prop or defaults to `false`.
+     * @returns {void} Does not return a value.
+     */
+    replaceThumbWithIcon(checked?: boolean) {
+        const isSwitchOn = checked ?? this.props.defaultChecked ?? false;
+        if (this.toggleRootRef.current) {
+            const thumbElement = (this.toggleRootRef.current as HTMLElement)?.querySelector('.ms-Toggle-thumb');
+
+            if (thumbElement) {
+                const style = ICON_STYLE.get(getIconStyleKey(this.props.size ?? UIToggleSize.Standard, isSwitchOn));
+                ReactDOM.render(
+                    <UIIcon iconName={isSwitchOn ? 'SwitchOn' : 'SwitchOff'} style={style} />,
```

Comment:
as there is enum, maybe `'SwitchOn'` -> `UiIcons.SwitchOn`?

Reply:
Done.